frontend/pages/calendar/index.vue

```diff
@@ -0,0 +1,312 @@
+          categoryData.forEach((category) => {
+            newData[category.id] = {
+              name: category.name,
+              color: category.color,
+            }
+          })
+          this.categoryData = newData
+        })
+        .catch((e) => {
+          console.log('categoryData >> 取得失敗')
+          console.log(e.message)
+        })
+    },
+    updateHeader() {
+      // タイトルとして使いたい情報を渡す
+      this.$nuxt.$emit('updateHeader', this.header.title)
+    },
+    async addTaskData(newTaskData) {
+      // サーバーに追加する
+      // this.$axios.post('/api/task/', newTaskData).then((response) => {
+
+      // })
+
+      // 【テスト】
+      const newId = this.makeTmpId('task')
+      newTaskData.id = newId
+
+      // IDBに登録する
+      this.$db.task.add(newTaskData).then(() => {
+        console.log('タスク追加 >> 成功')
+        this.getTasksFromDB()
+      })
+    },
+    deleteTaskData(taskId) {
+      // IDBから削除
+      this.$db.task.delete(taskId).then(() => {
+        console.log('task deleted.')
+        // タスクデータの再読み込み
+        this.getTasksFromDB()
+      })
+    },
+    updateTaskData(updatedData) {
+      const taskId = updatedData.id
+      delete updatedData.id
+
+      this.$db.task.update(taskId, updatedData).then(() => {
+        console.log('task updated.')
+        // タスクデータの再読み込み
+        this.getTasksFromDB()
+      })
+    },
+    updateCategoryData(updatedCategoryData) {
+      // カテゴリデータを更新
+      const targetId = updatedCategoryData.id
+      delete updatedCategoryData.id
+
+      this.$db.category.update(targetId, updatedCategoryData).then(() => {
+        console.log('category updated.')
+        // カテゴリデータの再読み込み
+        this.getCategoryDataFromDB()
+      })
+    },
+    addCategoryData(newCategoryData) {
+      // カテゴリデータを追加
+      // 【テスト】
+      const newId = this.makeTmpId('category')
+      newCategoryData.id = newId
+
+      this.$db.category.add(newCategoryData).then(() => {
+        console.log('category created.')
+        // カテゴリデータの再読み込み
+        this.getCategoryDataFromDB()
+      })
+    },
+    makeTmpId(type) {
+      // 【追加系api通信が新規IDを返すようになるまで使用】仮の新規IDを生成
+      let existIds = []
+      if (type == 'category') {
+        existIds = Object.keys(this.categoryData).map((id) => Number(id))
+      } else if (type == 'task') {
+        existIds = this.tasks.map((task) => task.id)
+      } else {
+        return
+      }
+
+      // 存在しているIDの最大値＋1
+      let maxId = null
+      if (existIds[0]) {
+        maxId = existIds.reduce((a, b) => {
+          return Math.max(a, b)
+        })
+        return maxId + 1
+      } else {
+        return 1
+      }
+    },
+    // 追加時の処理
+    onAdd(date) {
+      const now = new Date()
+      now.setDate(now.getDate() - 1)
+      const dt = new Date(date)
+      if (dt < now) {
+        this.snackbar = true
+      } else {
+        this.task.name = this.setItem[0].name
+        this.task.date = date
+        this.dialog = true
+        this.setItem = []
+      }
+    },
```

Comment:
ここは、nowという変数名にもかかわらず最終的に一日前の日付が入ってしまっています。
こうしたほうが分かりやすいと思うのですがどうですか？
```suggestion
    onAdd(date) {
      const now = new Date(this.date) // date--->todayの場合は this.today
      const targetDate = new Date(date)
      if (targetDate >= now) {
        this.task.name = this.setItem[0].name
        this.task.date = date
        this.dialog = true
        this.setItem = []
      } else {
        this.snackbar = true
      }
    },
```

Reply:
修正しました。